File: Programmers/42885.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int left = 0, right = people.length-1;

        ArrayList<Integer> peopleList = new ArrayList<>();
        for(int i = 0; i < people.length; i++) peopleList.add(people[i]);
        Collections.sort(peopleList);

        while(left < right){
            if(peopleList.get(left) + peopleList.get(right) > limit){
                right--;
                answer++;
            } else {
                left++;
                right--;
                answer++;
            }
        }

        if(left == right ) answer++;
        return answer;
    }
}

File: Programmers/42885.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool comp(int a, int b) {
    return a < b;
}

int solution(vector<int> people, int limit) {
    int left, right, answer = 0;

    reverse(people.begin(), people.end());
    sort(people.rbegin(), people.rend());	
    sort(people.begin(), people.end(), comp);
    
    left = 0;
    right = people.size()-1;
    
    while(left <= right){
        if(left == right ){
            answer++;
            break;
        }
        if(people[left] + people[right] > limit){
            right--; 
            answer++;
        } else {
            answer++;
            left++;
            right--;
        }  
    }
    
    return answer;
}